src/code/z_vr_box.c

```diff
@@ -1023,11 +1023,11 @@ void Skybox_Setup(GlobalContext* globalCtx, SkyboxContext* skyboxCtx, s16 skybox
     }
 }
 
-void Skybox_Init(GlobalContext* globalCtx, SkyboxContext* skyboxCtx, s16 skyboxId) {
+void Skybox_Init(GameState* state, SkyboxContext* skyboxCtx, s16 skyboxId) {
     skyboxCtx->unk_140 = 0;
     skyboxCtx->rot.x = skyboxCtx->rot.y = skyboxCtx->rot.z = 0.0f;
 
-    Skybox_Setup(globalCtx, skyboxCtx, skyboxId);
+    Skybox_Setup(state, skyboxCtx, skyboxId);
```

Comment:
ideally skybox_setup would be changed to expect state, but i couldnt get it to match myself. im pretty sure it should work though

Reply:
thats the reason why i leave it in its "wrong" state right now. the warning there will make sure we try to fix it in the right way. at least the way i think is probably correct, but that may be proven wrong

Reply:
I think the cast could just be here in `Skybox_Init`, not in `Skybox_Setup`

Reply:
im saying that i think setup should also be taking a state as its arg. it is incorrect for it to be expecting global context, cause it takes other states as well

Reply:
I just got it to match with a recast as well, so I think this is preferable

Reply:
I thought `Skybox_Init` was the function that was called by other game states and thus needed to take `state` as its argument, while `Skybox_Setup` is only called internally so it could continue taking `globalCtx`

Reply:
You can do that yeah, but conceptually it doesnt make sense. The function still gets used by gamestates other than play and will be passed pointers that should not be assumed to be global ctx

Reply:
I would argue `Skybox_Setup` itself is actually written expecting `globalCtx`, since it doesn't handle other game states well and it's basically a miracle that file select works with it as it is. Meanwhile the bug belongs in `Skybox_Init` which calls `Skybox_Setup` with a game state that might not be global context. So I think it makes more sense to cast in `Skybox_Init`.

Also in general I don't think we should have functions take game state as their argument unless they have to, so doing the cast in `Skybox_Init` would be better here imo

Reply:
I disagree that the bug is in `Skybox_Init`, that function still needs to be called for other game states. The issue imo is that one block of code in `Skybox_Setup` which lacks a check for the current gamestate and assumes that its global context